services/dynamic-sidecar/tests/unit/test_core_errors.py

```diff
@@ -0,0 +1,42 @@
+# pylint:disable=broad-exception-caught
+# pylint:disable=no-member
+
+from simcore_service_dynamic_sidecar.core.errors import (
+    UnexpectedDockerError,
+    VolumeNotFoundError,
+)
+from starlette import status
+
+
+def test_legacy_interface_unexpected_docker_error():
+    message = "some_message"
+    status_code = 42
+    try:
+        raise UnexpectedDockerError(  # noqa: TRY301
+            message=message, status_code=status_code
+        )
+    except Exception as e:
+        print(e)
+        assert e.status_code == status_code  # noqa: PT017
+        assert message in e.message  # noqa: PT017
+
+
+def test_legacy_interface_volume_not_found_error():
+    try:
+        volumes = [{}, {"Name": "a_volume"}]
+        volume_names = " ".join(v.get("Name", "UNKNOWN") for v in volumes)
+
+        raise VolumeNotFoundError(  # noqa: TRY301
+            volume_count=len(volumes),
+            source_label="some",
+            run_id="run_id",
+            volume_names=volume_names,
+            status_code=status.HTTP_404_NOT_FOUND,
+        )
+    except Exception as e:
+        print(e)
+        assert (  # noqa: PT017
+            e.message
+            == "Expected 1 got 2 volumes labels with source_label=some, run_id=run_id: Found UNKNOWN a_volume"
```

Comment:
TIP: kind reminder about an iterating topic in our reviews. 

IMO it does not really pay back having a test that checks error messages verbatim
I would rather capture a base error and check the **context** variables that you are raising with your exception. Look at the functions in the base class and try to use them

Reply:
While I do understand your point and I know it makes sense, here we have bit of a special case.

The idea is to enforce exactly the same interface from one version to the next. Both errors contain attributes called `message` and `status_code`, are used elsewhere and we rely on the fact that they exist as they are currently provided.